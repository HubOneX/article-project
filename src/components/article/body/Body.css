.article-body {
  padding: 0 30px 30px;
  margin: 0;

  & h2 {
    font-size: 26px;
    margin: 5px 0;
  }

  & p {
    font-size: 22px;
    text-align: justify;
  }

  @media screen and (max-width: 480px) {
    padding: 0 20px 20px;

    & h2 {
      font-size: 22px;
    }

    & p {
      font-size: 18px;
    }
  }
}
