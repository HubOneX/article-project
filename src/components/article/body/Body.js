import NotLoaded from "../notLoaded/NotLoaded";
import "./Body.css";

const Body = ({ bodyElements }) => {
  if (!bodyElements || bodyElements.length === 0) {
    return <NotLoaded />;
  }

  return (
    <div className="article-body">
      {bodyElements.map((element) => (
        <div
          dangerouslySetInnerHTML={{ __html: element }}
          key={bodyElements.indexOf(element)}
        />
      ))}
    </div>
  );
};

export default Body;
