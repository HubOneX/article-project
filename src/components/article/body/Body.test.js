import renderer from "react-test-renderer";
import Body from "./Body";
import { render, screen } from "@testing-library/react";

describe("Body", () => {
  it("should match the snapshot with correctly provided array of html elements", () => {
    const body = renderer.create(
      <Body
        bodyElements={[
          "<h2>This is a sub-heading</h2>",
          "<p>This is a paragraph!</p>",
        ]}
      />
    );

    expect(body).toMatchSnapshot();
  });

  it("should render information if bodyElements ARRAY is empty", () => {
    render(<Body bodyElements={[]} />);
    const information = screen.getByText(
      "Sorry, we couldn't load this article. Please try again later."
    );

    expect(information).toBeInTheDocument();
  });

  it("should render information if bodyElements is an empty STRING", () => {
    render(<Body bodyElements="" />);
    const information = screen.getByText(
      "Sorry, we couldn't load this article. Please try again later."
    );

    expect(information).toBeInTheDocument();
  });

  it("should render information if bodyElements is an not provided AT ALL", () => {
    render(<Body />);
    const information = screen.getByText(
      "Sorry, we couldn't load this article. Please try again later."
    );

    expect(information).toBeInTheDocument();
  });
});
