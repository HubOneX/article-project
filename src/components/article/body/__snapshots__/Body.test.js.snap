// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should match the snapshot with correctly provided array of html elements 1`] = `
<div
  className="article-body"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h2>This is a sub-heading</h2>",
      }
    }
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>This is a paragraph!</p>",
      }
    }
  />
</div>
`;
